### DISCUSSION TITLE
Send error codes for errors

### DISCUSSION THREAD

**Discussion by Jonny1987** [supabase/supabase#21479] [Feature Requests]
# Chore
Give each error an error code

## Describe the chore
Currently many errors have the same status code (eg 400) and the only information about the error is the message. This means that in my code, I need to write a condition for this error message if I want to check for it, but this means if the error message changes then my code will no longer work.

For example, the supabase.auth.signInWithPassword() function in flutter raises error with the message  "Email not confirmed" or "Invalid login credentials" in the corresponding situations. But the status code for both is 400 so I can only check for this message exactly as written.

It would be better if the response also sent an error code so that the condition could be made on this, without risking any problems if the error message gets changed in the future.

## Additional context

Add any other context or screenshots that help clarify the task.

**Comment by IcedTea2K** [comment]
A response with 400 code generally means a Bad Request to the server. It's a standard way to communicate over the internet. Though, I totally agree with the idea. Maybe having an internal error code attached with the body might be nice.

Would like to hear other people's ideas on this

**Comment by encima** [comment]
As this is a more "open" issue, I would suggest moving this to a discussion to discuss the best approach that would be consistent, least impactful for existing implementations and logical for new users.

**Comment by dshukertjr** [comment]
Error codes are being implemented right now. Here is a PR for the JS SDK, and we will being it soon to Flutter as well. 
https://github.com/supabase/gotrue-js/pull/855

**Reply by ymerdrengene** [reply]
Any status on when it will be available for Flutter?

**Comment by bahaafathi** [comment]
any update?
